_posts/2018/2018-02-18-avoid-writing-sql-when-using-activerelation.md

```diff
@@ -0,0 +1,67 @@
+---
+title: 'Avoid Writing SQL When Using ActiveRelation'
+description: 'It's fine, but prefer not to if you can.'
+layout: article
+category: ruby
+image:
+  base: '2018/avoid-writing-sql-when-using-activerelation'
+  alt: 'Coding SQL'
+  source: 'https://unsplash.com/photos/fPkvU7RDmCo'
+  credit: 'Caspar Rubin'
+---
+
+ActiveRelation, which powers the searching and querying engine of ActiveRecord, is a powerful and flexible tool.
+
+## Instead of...
+
+...writing literal SQL snippets inside an ActiveRelation `#where` method.
+
+```ruby
+class Screenshot < ApplicationRecord
+  scope :visible, -> where('automated_screen_grab = ? AND hidden_at IS NULL', true)
+end
+```
+
+
+## Use...
+
+...the hash-style syntax.
+
+```ruby
+class Screenshot < ApplicationRecord 
+  scope :visible, -> where(automated_screen_grab: true, hidden_at: nil)
+end
+```
+
+
+## Why?
+
+Both methods are valid and the “string with interpolated values” version was the _only_ way before ActiveRelation. The hash-style gives you more flexibility.
```

Comment:
*However, the hash-style gives you more...

Reply:
Maybe change: "“string with interpolated values” to: 'first'...or 'SQL' version to slightly reduce cognitive processing effort. :P